### PR TITLE
Associations

### PR DESCRIPTION
**Pull request by nunegabriel**
In this pull request I did the following:

- [ ] Created a class Classroom with Setter and getter for @label 
- [ ]  Created the has-many/belongs-to relationship between Classroom and Student.
- [ ] Created a class Book with the following:
@title and @author instance variables, should be initialized in the constructor.
- [ ] Create a class Rental with the following:
@date instance variable, should be initialized in the constructor.
Setter and getter for @date (remember about attr_accessor).
- [ ]  Created the has-many side of Book and Rental (a book has many rentals).
- [ ] Created the belongs-to side of Rental and Book (a rental belongs to a book).
- [ ] Created the has-many side of Person and Rental (a person has many rentals).
- [ ] Created the belongs-to side of Rental and Person (a rental belongs to a person).
- [ ] Modified the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -9,6 +12,7 @@ def initialize(_name, age, _parent_permission)
     @name = 'Unknown'
     @age = age
     @parent_permission = true
+    @rentals = []
```
Comment (Ghiftee) [inline person.rb:15]: - [ ] Kindly create a method for adding rentals  e.g
```ruby
def add_rental(book, date)
    Rental.new(date, book, self)
  end
```

File: rental.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class Rental
+  attr_accessor :date, :person, :book
+
+  def initialize(date)
```
Comment (Ghiftee) [inline rental.rb]: - [ ] The constructor of Rental should be modified so Book and Person are set in it to break the many-to-many relationship. e.g
`def initialize(date, book, person)`
You can read more about this [here](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship)

**Review by Ghiftee — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️♻️
Hello @nunegabriel  👋🏽 

Good job working on the project so far! 
To highlight 👍🏽 
✔️ You created classes
✔️ Associations were set up
✔️ Your README is professional

However, there are some issues that you need to work on to go to the next project but you are almost there!



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
- For best practice, I suggest writing your commit messages in lowercase letters and capitalizing only the first letter.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by mikethreels — APPROVED** [review]
Hi Gabby, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- linters are passing
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**